devito/mpi/distributed.py

```diff
@@ -19,9 +19,24 @@
 # Do not prematurely initialize MPI
 # This allows launching a Devito program from within another Python program
 # that has *already* initialized MPI
-import mpi4py
-mpi4py.rc(initialize=False, finalize=False)
-from mpi4py import MPI  # noqa
+try:
+    import mpi4py
+    mpi4py.rc(initialize=False, finalize=False)
+    from mpi4py import MPI  # noqa
+except ImportError:
+    class MPI(object):
+        COMM_NULL = None
+
+        @classmethod
+        def Is_initialized(cls):
+            return False
+
+        def _sizeof(comm):
+            return None
```

Comment:
better to return 0

Reply:
I prefer None, size of undefined shouldn't be zero, but that's just me